app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'AuthorId'
+  has_many :likes
+  has_many :comments
+
+  def update_post_counter(user)
+    post_count = Post.where(AuthorId: user.id).count
+    user.update(PostCounter: post_count)
+  end
```

Comment:
[ OPTIONAL] 
- You can use one of [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private. See as example below 
`after_save :update_post_counter` 
`private` 
`def update_post_counter`            `author.increment!(:posts_counter)` 
`end`